cpp/memilio/epidemiology/lct_populations.h

```diff
@@ -0,0 +1,212 @@
+            return get_first_index_group<Group - 1>() + std::tuple_element_t<Group - 1, tupleLctStates>::Count;
+        }
+    }
+    /**
+     * @brief Returns an Eigen copy of the vector of populations. 
+     * This can be used as initial conditions for the ODE solver.
+     * @return Eigen::VectorXd of populations.
+     */
+    inline Vector<FP> get_compartments() const
+    {
+        return m_y.array().template cast<FP>();
+    }
+
+    /**
+     * @brief Returns the total population of a group.
+     * @tparam group The group for which the total population should be calculated.
+     * @return Total population of the group.
+     */
+    template <size_t group>
+    ScalarType get_group_total() const
+    {
+        return m_y.array()
+            .segment(get_first_index_group<group>(), std::tuple_element_t<group, tupleLctStates>::Count)
+            .sum();
+    }
+
+    /**
+     * @brief Returns the total population of all compartments and groups.
+     * @return Total population.
+     */
+    ScalarType get_total() const
+    {
+        return m_y.array().template cast<ScalarType>().sum();
+    }
+
+    /**
+     * @brief Checks whether all compartments have non-negative values. 
+     * This function can be used to prevent slighly negative function values in compartment sizes that came out
+     * due to roundoff errors if, e.g., population sizes were computed in a complex way.
+     *
+     * Attention: This function should be used with care. It can not and will not set model parameters and 
+     *            compartments to meaningful values. In most cases it is preferable to use check_constraints,
+     *            and correct values manually before proceeding with the simulation.
+     *            The main usage for apply_constraints is in automated tests using random values for initialization.
+     *
+     * @return Returns true if one (or more) constraint(s) were corrected, otherwise false.
+    */
+    bool apply_constraints()
+    {
+        bool corrected = false;
+        for (int i = 0; i < m_y.array().size(); i++) {
+            if (m_y.array()[i] < 0) {
+                log_warning("Constraint check: Compartment size {:d} changed from {:.4f} to {:d}", i, m_y.array()[i],
+                            0);
+                m_y.array()[i] = 0;
+                corrected      = true;
+            }
+        }
+        return corrected;
+    }
+
+    /**
+     * @brief Checks whether all compartments have non-negative values and logs an error if constraint is not satisfied.
+     * @return Returns true if one or more constraints are not satisfied, false otherwise.
+     */
+    bool check_constraints() const
+    {
+        if ((m_y.array() < 0.).any()) {
+            log_error("Constraint check: At least one compartment size is smaller {}.", 0);
+            return true;
+        }
+        return false;
+    }
+
+private:
+    /**
+     * @brief Sets recursively the total number of (sub-)compartments in all groups. 
```

Comment:
```suggestion
     * @brief Sets recursively the total number of (sub-)compartments over all groups. 
```
Here, it was not clear to me if this is done for every group oryou count the number of subcompartments when considering all groups.